### PR TITLE
torch/tfjs: Translate select to gather.

### PR DESCRIPTION
**Pull request by juharris**
## Description
Translate PyTorch's gather to Tensorflow.js' gather. I am trying to write a Plan in Python with PyTorch that uses `tensor.select` and I would like it translated so that I can use `syft,js`. Did I do this right?

## How has this been tested?
Added tests.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by vvmnnnkv — APPROVED** [review]

**Review by Nolski — APPROVED** [review]
Just fix the merge conflicts and I'll merge :)

**Comment by juharris** [comment]
@Nolski I resolved the conflicts.
